bitcoind-rpc/src/main/scala/org/bitcoins/rpc/client/common/BitcoindRpcClient.scala

```diff
@@ -44,6 +53,56 @@ class BitcoindRpcClient(val instance: BitcoindInstance)(
   require(version == BitcoindVersion.Unknown || version == instance.getVersion,
           s"bitcoind version must be $version, got ${instance.getVersion}")
 
+  // Chain Query Api
+
+  /** Gets the height of the given block */
+  override def getBlockHeight(
+      blockHash: DoubleSha256DigestBE): Future[Option[Int]] = {
+    getBlockHeader(blockHash).map(header => Some(header.height))
+  }
+
+  /** Gets number of confirmations for the given block hash */
+  override def getNumberOfConfirmations(
+      blockHash: DoubleSha256DigestBE): Future[Option[Int]] = {
+    getBlockHeader(blockHash).map(header => Some(header.confirmations))
+  }
+
+  /** Gets the number of compact filters in the database */
+  override def getFilterCount: Future[Int] =
+    Future.failed(
+      throw new UnsupportedOperationException(
+        s"bitcoind ${instance.getVersion} does not support block filters"))
+
+  /** Returns the block height of the given block stamp */
+  override def getHeightByBlockStamp(blockStamp: BlockStamp): Future[Int] =
+    blockStamp match {
+      case blockHeight: BlockStamp.BlockHeight =>
+        Future.successful(blockHeight.height)
+      case blockHash: BlockStamp.BlockHash =>
+        getBlockHeader(blockHash.hash).map(_.height)
+      case blockTime: BlockStamp.BlockTime =>
+        Future.failed(
+          new UnsupportedOperationException(s"Not implemented: $blockTime"))
+    }
+
+  override def getFiltersBetweenHeights(
+      startHeight: Int,
+      endHeight: Int): Future[Vector[ChainQueryApi.FilterResponse]] =
+    Future.successful(Vector.empty)
+
+  /** Gets the block height of the closest block to the given time */
+  override def epochSecondToBlockHeight(time: Long): Future[Int] =
+    Future.failed(
+      new UnsupportedOperationException(
+        "epochSecondToBlockHeight is not supported by bitcoind"))
+
+  // Node Api
+
+  override def broadcastTransaction(transaction: Transaction): Future[Unit] =
+    sendRawTransaction(transaction).map(_ => ())
+
+  override def downloadBlocks(
+      blockHashes: Vector[DoubleSha256Digest]): Future[Unit] = FutureUtil.unit
```

Comment:
Can we not retrieve full blocks from disk?

Reply:
We can, but that won't actually do anything for the callback because it needs to come from the network to be passed through the data message handler, I might be able to fix this actually. Will take a second look.

Reply:
So, it's not actually possible to call or set the wallet callbacks without having the wallet actually present. Maybe this should be changed to return a `Future[Vector[Block]]`

Reply:
Yeah, I think we should consider re-working this as it doesn't make much sense. Maybe file an issue and we will go with this implementation for now?

Reply:
https://github.com/bitcoin-s/bitcoin-s/issues/1497

Reply:
Oh, I made #1487